### PR TITLE
Fix units in sizeof_fmt

### LINKED ISSUE #12961: Units in `mne.utils.sizeof_fmt()` incorrect
The helper function `mne.utils.sizeof_fmt()` creates a string of a given number of bytes in a human-readable format (with appropriate prefixes). However, it uses base 2, so the units should be kiB, MiB, GiB, ... and not kB, MB, GB, ... (which are base 10).

We should either change the displayed units or the base. Which one would you prefer?

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #12961.

### CODE REVIEW

**Review by agramfort — COMMENTED** [review]
to me MB and GB is much more standard. eg https://www.vaia.com/en-us/explanations/computer-science/computer-organisation-and-architecture/units-of-data-storage/#:~:text=The%20different%20units%20of%20data%20storage%20in%20computer%20science%20include,%2C%20and%20Yottabyte%20(YB).

**Comment by cbrnr** [comment]
Yes, but then we have to switch to base 1000, because currently we are using base 1024 (e.g. 1 kB = 1000 bytes, but 1 kiB = 1024 bytes). I'm fine either way.

**Comment by drammock** [comment]
This was [discussed before](https://github.com/mne-tools/mne-python/pull/12787#issuecomment-2288868681) in regards to the sys_info output and we settled on GiB so let's be consistent and stick with that here.

File: doc/changes/devel/12962.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix displayed units in :func:`mne.utils.sizeof_fmt` to correctly use kiB, MiB, GiB, and so on, by `Clemens Brunner`_.
```
Comment (drammock) [inline doc/changes/devel/12962.bugfix.rst:1]:
```suggestion
Fix displayed units in :func:`mne.utils.sizeof_fmt` to correctly use KiB, MiB, GiB, and so on, by `Clemens Brunner`_.
```

**Review by drammock — APPROVED** [review]
LGTM, just one capitalization fix in the changelog entry

**Review by agramfort — APPROVED** [review]

**Comment by cbrnr** [comment]
I'll fix the tests this afternoon!

File: doc/changes/devel/12962.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix displayed units in :func:`~mne.utils.misc.sizeof_fmt` to correctly use KiB, MiB, GiB, and so on, by `Clemens Brunner`_.
```
Comment (larsoner) [inline doc/changes/devel/12962.bugfix.rst:1]: We can only crossref things that are documented in our API reference, and `mne.utils` is not. So I would refer to the more public / end-user effect instead
```suggestion
Fix displayed units in representations of classes such as :class:`mne.io.Raw` to correctly use KiB, MiB, GiB, and so on, by `Clemens Brunner`_.
```

**Pull request merged**